components/layout_2020/table/construct.rs

```diff
@@ -287,26 +287,16 @@ impl TableBuilder {
     }
 
     /// It's possible to define more table columns via `<colgroup>` and `<col>` elements
-    /// than actually exist in the table. In that case, remove these bogus columns
-    /// to prevent using them later in layout.
-    fn remove_extra_columns_and_column_groups(&mut self) {
-        let number_of_actual_table_columns = self.table.size.width;
-        self.table.columns.truncate(number_of_actual_table_columns);
-
-        let mut remove_from = None;
-        for (group_index, column_group) in self.table.column_groups.iter_mut().enumerate() {
-            if column_group.track_range.start >= number_of_actual_table_columns {
-                remove_from = Some(group_index);
-                break;
-            }
-            column_group.track_range.end = column_group
-                .track_range
-                .end
-                .min(number_of_actual_table_columns);
-        }
-
-        if let Some(remove_from) = remove_from {
-            self.table.column_groups.truncate(remove_from);
+    /// than actually exist in the table. In that case, increase the size of the table.
+    ///
+    /// However, if the table has no row nor row group, remove the extra columns instead.
+    /// This matches WebKit, and some tests require it, but Gecko and Blink don't do it.
+    fn do_extra_columns_and_column_groups_fixup(&mut self) {
```

Comment:
How about `remove_columns_and_column_groups_from _empty_table`?

Reply:
Or perhaps `remove_extra_columns_and_column_groups_if_table_has_no_rows`.

Reply:
But that's like a corner case that doesn't even match Geck and Blink. Usually what this function will do is increase the width for the extra columns:

```rs
self.table.size.width = self.table.size.width.max(self.table.columns.len());
```

Reply:
How about `adjust_table_geometry_for_columns_and_colgroups` here where table geometry can be either the number of cells or the number of columns / colgroups?